tests/installation/hostname_inst.pm

```diff
@@ -27,6 +27,8 @@ sub run {
     # 'install' is the default hostname if no hostname is get from environment
     my $NICTYPE_USER_OPTIONS      = get_var('NICTYPE_USER_OPTIONS', 'install');
     my $expected_install_hostname = ($NICTYPE_USER_OPTIONS =~ s/hostname=//r);
+    # but, we may expect a different hostname
+    $expected_install_hostname = get_var('EXPECTED_INSTALL_HOSTNAME', 'install');
```

Comment:
Hmm not sure if a record_info should be added in the case that both are set, just to avoid confusion if a problem arises

Reply:
`NICTYPE_USER_OPTIONS` just means that a fake DHCP server pushes a hostname to the SUT.
If `EXPECTED_INSTALL_HOSTNAME` is not set, then `NICTYPE_USER_OPTIONS` is the expected name.
But, there are some cases, like the scenario that Rodion showed, were we expect something different (linuxrc, hostname, a real DHCP server pushing a hostname)
So, `EXPECTED_INSTALL_HOSTNAME` has precedence if set.

So, I assume the confusion is about to make the precedence explicit. Adding to doc.

Reply:
FTR: https://openqa.suse.de/tests/4010605